### PR TITLE
Feat: Adds ue ip to action result

### PR DESCRIPTION
**Pull request by sanchezfdezjavier**
# Description

Adds ue ip to `attach-ue` action result.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -134,3 +135,18 @@ def ip_from_iface(subnet: str) -> Optional[str]:
...
+
+def get_iface_ip_address(iface: str) -> Optional[str]:
+    """Get the UE IP address.
+
+    Args:
+        iface: The interface name.
+
+    Returns:
+        str: UE IP address
+    """
```
Comment (saltiyazan) [inline src/utils.py:148]: I think the name of the function and the docstring don't really match.
Could you please change the docstring so the function is more general?
```suggestion
    """Get the IP address of the given interface.

    Args:
        iface: The interface name.

    Returns:
        str: IP address
    """
```

File: README.md [hunk 2]
```diff
@@ -58,3 +58,7 @@ juju run-action <unit> remove-default-gw --wait
 
 - **lte-core**: The LTE core interface is used to connect to a 4G/LTE core network via its MME IPv4 address.
 
+## Sample charms implementing the interface
+
+- [srs-enb-ue-operator](https://github.com/charmed-osm/srs-enb-ue-operator)
+- [magma-access-gateway-operator](https://github.com/canonical/magma-access-gateway-operator)
```
Comment (saltiyazan) [inline README.md:64]:
```suggestion
- Requirer charm: [srs-enb-ue-operator](https://github.com/charmed-osm/srs-enb-ue-operator)
- Provider charm: [magma-access-gateway-operator](https://github.com/canonical/magma-access-gateway-operator)
```

File: src/charm.py [hunk 3]
```diff
@@ -209,7 +210,18 @@ def _on_attach_ue_action(self, event: ActionEvent) -> None:
...
-        event.set_results({"status": "ok", "message": "Attached successfully"})
+        if ue_ip := get_iface_ip_address("tun_srsue"):
+            event.set_results(
+                {"status": "ok", "message": "Attached successfully", "ue-ipv4": ue_ip}
+            )
+        else:
+            event.set_results(
+                {
+                    "status": "ok",
+                    "message": "Attached successfully",
```
Comment (saltiyazan) [inline src/charm.py:221]: is successful in this case?

File: README.md [hunk 4]
```diff
@@ -58,3 +58,7 @@ juju run-action <unit> remove-default-gw --wait
 
 - **lte-core**: The LTE core interface is used to connect to a 4G/LTE core network via its MME IPv4 address.
 
+## Sample charms implementing the interface
```
[the remark this answers is not in the document]
Comment (sanchezfdezjavier) [reply README.md]: Done.

File: src/utils.py (hunk 1 above)
Comment (sanchezfdezjavier) [reply src/utils.py]: Done. Copilot doing copilot things 😅

File: README.md (hunk 2 above)
Comment (sanchezfdezjavier) [reply README.md]: Nice comment. Done.

File: src/charm.py (hunk 3 above)
Comment (sanchezfdezjavier) [reply src/charm.py:221]: I wanted to discuss this, shouldn't we merge the 3 fields into one? Something like:
When successful
```python
{
	"status": "ok",
"message": "Attached successfully",
                    "ue-ipv4": "No UE ip found, please make sure the interface is up.",
                }
```

**Review by saltiyazan — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -58,3 +58,7 @@ juju run-action <unit> remove-default-gw --wait
 
 - **lte-core**: The LTE core interface is used to connect to a 4G/LTE core network via its MME IPv4 address.
 
+### Sample charms implementing the interface
```
Comment (gruyaume) [inline README.md:61]: I don't think we need this here, if it's anywhere it should be in the interface project

File: src/charm.py [hunk 6]
```diff
@@ -209,7 +210,15 @@ def _on_attach_ue_action(self, event: ActionEvent) -> None:
         service_restart(SRS_UE_SERVICE)
         self._stored.ue_attached = True
         self.unit.status = self._get_current_status()
-        event.set_results({"status": "ok", "message": "Attached successfully"})
+        if ue_ip := get_iface_ip_address("tun_srsue"):
+            event.set_results({"message": "Attached successfully.", "ue-ipv4": ue_ip})
+        else:
+            event.set_results(
```
Comment (gruyaume) [inline src/charm.py:216]: should use event.fail instead

Comment (sanchezfdezjavier) [reply src/charm.py]: Fixed.

File: README.md (hunk 5 above)
Comment (sanchezfdezjavier) [reply README.md]: Oh, for sure. Removed.

File: src/charm.py (hunk 3 above)
Comment (sanchezfdezjavier) [reply src/charm.py]: I think I didn't get it at first, I change it a bit. Let me know what you think.

**Review by gruyaume — APPROVED** [review]

**Pull request merged**